pkg/operator/utils/topology.go

```diff
@@ -14,3 +19,19 @@ func GetTopologyCategories(clusterCSIDriver *opv1.ClusterCSIDriver) []string {
 	}
 	return []string{}
 }
+
+func GetDatacenters(config *vsphere.VSphereConfig) ([]string, error) {
+	datacenters := []string{config.Workspace.Datacenter}
+
+	virtualCenterIPs := sets.StringKeySet(config.VirtualCenter)
+
+	if len(virtualCenterIPs) != 1 {
+		return nil, fmt.Errorf("cloud config must define a single VirtualCenter")
+	}
+
+	virtualCenterIP := virtualCenterIPs.List()[0]
+	if virtualCenterConfig, ok := config.VirtualCenter[virtualCenterIP]; ok {
+		datacenters = strings.Split(virtualCenterConfig.Datacenters, ",")
+	}
+	return datacenters, nil
```

Comment:
Can we unify with the common logic from `targetconfigcontroller.go`?

Reply:
fixed